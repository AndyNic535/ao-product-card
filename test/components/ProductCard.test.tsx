import React from "react";
import render from "react-test-renderer";
import { ProductCard } from '../../src/components';
import { product1 } from '../data/product';


const { act } = render;


describe( 'Pruebas del componente <ProductCard />', () => {

    test( 'Debe de mostrar el componente correctamente.', () => {

        const wrapper = render.create(

            <ProductCard product={ product1 } >
                {
                    () => (

                        <h1>Product Card</h1>

                    )
                }
            </ProductCard>

        );

        // console.log( wrapper.toJSON() );

        expect( wrapper.toJSON() ).toMatchSnapshot();

    });

    test( 'Debe de mostrar el contador con el valor 0 y el boton para incrementar en 1', () => {

        const wrapper = render.create(

            <ProductCard product={ product1 } >
                {
                    ({ count, increaseBy }) => (
                        <>

                            <h1>Product Card</h1>
                            <span>{ count }</span>
                            <button onClick={ () => increaseBy( +1 ) } >+ 1</button>

                        </>
                    )
                }
            </ProductCard>

        );

        // console.log( wrapper.toJSON() );

        expect( wrapper.toJSON() ).toMatchSnapshot();

    });

    test( 'Debe de incrementar el contador en 1.', () => {

        const wrapper = render.create(

            <ProductCard product={ product1 } >
                {
                    ({ count, increaseBy }) => (
                        <>

                            <h1>Product Card</h1>
                            <span>{ count }</span>
                            <button onClick={ () => increaseBy( +1 ) } >+ 1</button>

                        </>
                    )
                }
            </ProductCard>

        );

        let tree = wrapper.toJSON();

        act( () => {

            ( tree as any ).children[ 2 ].props.onClick();

        });

        tree = wrapper.toJSON();

        expect( ( tree as any ).children[ 1 ].children[ 0 ] ).toBe( '1' );

    });

});