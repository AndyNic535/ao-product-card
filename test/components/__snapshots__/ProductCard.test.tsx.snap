// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <ProductCard /> Debe de mostrar el componente correctamente. 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
</div>
`;

exports[`Pruebas del componente <ProductCard /> Debe de mostrar el contador con el valor 0 y el boton para incrementar en 1 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    + 1
  </button>
</div>
`;
